### DISCUSSION TITLE
How to minimize SASS modules output ?

### DISCUSSION THREAD

**Discussion by Hellaeh** [vercel/next.js#22269] [Help] [answered]
Hello. I'm using css modules with sass. The problem is it's not minified... Even in production i still get `<div class="container_container__j_owi">` or  `<nav class="navbar_nav__2KqJF>`. How to minimize\obfuscate classes and ids ?

Ideally, i'd like to get a couple of chars output, like "aa" or "a1" instead of "container_container__j_owi".

**Comment by Hellaeh** [comment, marked as answer]
Nvm. After hours of searching i found someone else solution and came up with this. next.config.js:

```
const path = require('path');

const pathCache = {};
const generatorCache = {};

const getRandomPseudoHash = () =>
	Math.random()
		.toString(32)
		.slice(-2)
		.replace(/^\d/, '_')
		.replace(/\w/g, char => (Math.random() > 0.5 ? char.toUpperCase() : char));

module.exports = {
	webpack: nextConfig => {
		nextConfig.module.rules[1].oneOf.forEach(moduleLoader => {
			Array.isArray(moduleLoader.use) &&
				moduleLoader.use.forEach(x => {
					if (x.loader.match(/(\/|\\)css-loader/)) {
						x.options.modules.getLocalIdent = (context, _, className) => {
							const resPath = context?.resourcePath;
							if (resPath) {
								const pathName = `${resPath}_${className}`;
								if (pathCache[pathName]) return pathCache[pathName];
								else {
									let buf = getRandomPseudoHash();
									while (generatorCache[buf]) {
										buf = getRandomPseudoHash();
									}
									generatorCache[buf] = buf;
									pathCache[pathName] = buf;
									return buf;
								}
							}
						};
					}
				});
		});
		return nextConfig;
	},

	sassOptions: {
		includePaths: [path.join(__dirname, 'src/styles')],
	},
};
```
Two char should be enough for ~2000 cases. But compilation time will increase dramatically near cap, so just add more chars.

**Reply by austinbiggs** [reply]
Thanks for sharing this!

**Reply by Hellaeh** [reply]
Glad to help. Just be careful and avoid infinite loop, and you probly want to disable it for development.

**Reply by dsacramone** [reply]
What sort of perf / size benefit did you see from this?

**Reply by Hellaeh** [reply]
It totally depends on your project. Performance difference should be none at all. And bundle size depends, on: 

- you're reusing css classes\ids or create new for every element. 
- project size overall

In my case (reusing)  it's only ~1kb difference atm, but it's solid 15%(css only). 

I'm in the middle of development tho. Still `aa < container_container__j_owi` simple.
